### ISSUE TITLE
在训练过程中显存明显增加

### ISSUE THREAD

**Issue by tricksterANDthug** [PaddlePaddle/Paddle#42197]
### 请提出你的问题 Please ask your question

for epoch in range(epoch_num):
        print(f'Epoch {epoch + 1} / {epoch_num}')
        model.train()
        for batch_id, data in enumerate(train_loader()):
            loss, acc, miou = train(data, model)
            # save log
            writer.add_scalar('train/loss', step=train_step, value=loss.numpy()[0])
            writer.add_scalar('train/acc', step=train_step, value=acc)
            writer.add_scalar('train/miou', step=train_step, value=float(miou))
            train_step += 1
            # print loss and acc
            print({'loss': loss.numpy()[0], 'acc': acc, 'miou': miou})
            loss.backward()
            optimizer.step()
            optimizer.clear_grad()

        model.eval()
        for batch_id, data in enumerate(val_loader()):
            loss, acc, miou = train(data, model)
            writer.add_scalar('val/loss', step=val_step, value=loss.numpy()[0])
            writer.add_scalar('val/acc', step=val_step, value=acc)
            writer.add_scalar('val/miou', step=val_step, value=float(miou))
            val_step += 1
            # print loss and acc
            print({'loss': loss.numpy()[0], 'acc': acc, 'miou': miou})
设置的batch_size = 15
在model.train() 显存占用是8.8g
在model.eval()显存直接增加到15.8g，机会增加了一倍
这个合理吗

**Comment by xyzkk3** [comment]
同样的问题，调用eval后显存爆增，溢出报错，如果不调用eval一直处于train状态，能将几个epoch跑完

**Comment by xyzkk3** [comment]
> 同样的问题，调用eval后显存爆增，溢出报错，如果不调用eval一直处于train状态，能将几个epoch跑完

找到问题了，用的是2.3版本的paddle，eval之后依然会计算梯度，加了个paddle.no_grade就不会再报错了
